editor.py

```diff
@@ -66,25 +72,41 @@ def editor(initial_contents=None, filename=None, editor=None):
 
       editor
         The path to an editor to call.  If None, use editor.default_editor()
+        If None, use editor.default_editor().
+        If `editor` is a string, it will be split using `shlex`
+        If it is a list or tuple of strings, it will be used as is.
     """
     editor = editor or default_editor()
-    if not filename:
-        with tempfile.NamedTemporaryFile(mode='r+', suffix='.txt') as fp:
-            if initial_contents is not None:
-                fp.write(initial_contents)
-                fp.flush()
-
-            subprocess.call([editor, fp.name])
-
-            fp.seek(0)
-            return fp.read()
-
-    path = Path(filename)
-    if initial_contents is not None:
-        path.write_text(initial_contents)
-
-    subprocess.call([editor, filename])
-    return path.read_text()
+    if isinstance(editor, str):
+        editor = shlex.split(editor)
```

Comment:
I'm starting to think differently, now that I see this `shlex` change...

Why should we bother about splitting the string and detecting shell syntax altogether? Why not just pass a space-separated string to `Popen` with `shell=True`? :thinking:

Reply:
That's also an excellent question!  Thanks for all these good ideas.

---

The question hinges on this - is there some other reason you might want `shell=True` or `False` independently of the argument?

If the answer to this is "yes", then we should let people set `shell=` and then use code like this (from another project of mine)

        if is_str and not shell:
            cmd = shlex.split(cmd)
        if not is_str and shell:
            cmd = shlex.join(cmd)

If not, then we should use your solution.

Either way the code needs to change!

I'm going to go and find out the answer to this question.

---

I really appreciate thoughtful reviews.  :-)